labtool2.0/src/components/pages/ReviewStudent.js

```diff
@@ -43,12 +50,43 @@ export class ReviewStudent extends Component {
     }
   }
 
+  toggleCheckbox = name => async e => {
+    this.props.toggleCheck(name)
+  }
+
+  copyChecklistOutput = async e => {
+    e.preventDefault()
+    this.reviewPointsRef.current.inputRef.value = e.target.points.value
+    /* The below line is as hacky as it is because functional elements cannot directly have refs.
+    * This abomination somehow accesses a textarea that is a child of a div that holds the ref.
+    */
+    this.reviewTextRef.current.children[0].children.comment.value = e.target.text.value
+  }
+
   render() {
     //this.props.ownProps.studentInstance is a string, therefore casting to number.
     const studentData = this.props.courseData.data.filter(dataArray => dataArray.id === Number(this.props.ownProps.studentInstance))
     //this.props.weekNumber is a string, therefore casting to number.
     const weekData = studentData[0].weeks.filter(theWeek => theWeek.weekNumber === Number(this.props.ownProps.weekNumber))
     const checkList = this.props.selectedInstance.checklists.find(checkl => checkl.week == this.props.ownProps.weekNumber)
+    let checklistOutput = ''
+    let checklistPoints = 0
+    if (checkList) {
+      Object.keys(checkList.list).forEach(cl => {
+        checkList.list[cl].forEach(row => {
+          const addition = this.props.weekReview.checks[row.name] ? row.textWhenOn : row.textWhenOff
+          if (addition) checklistOutput += addition + '\n\n'
```

Comment:
It would be better if the feedback is one line without line breaks to minimize the editing effort. Then we would need be a check for first letter in upper case and a dot at the end of each addition to maintain the readability. This can also be fixed later.